hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonRecordReader.java

```diff
@@ -50,6 +50,8 @@
   protected QueryExecutor queryExecutor;
   private InputMetricsStats inputMetricsStats;
 
+  protected boolean isSearchMode = false;
```

Comment:
It is not so intuitive to understand this, better to add a subclass to override `close` function